x/stake/keeper/query_utils.go

```diff
@@ -0,0 +1,101 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/stake/types"
+)
+
+// Return all validators that a delegator is bonded to. If maxRetrieve is supplied, the respective amount will be returned.
+func (k Keeper) GetDelegatorValidators(ctx sdk.Context, delegatorAddr sdk.AccAddress,
+	maxRetrieve uint16) (validators []types.Validator) {
+	validators = make([]types.Validator, maxRetrieve)
+
+	store := ctx.KVStore(k.storeKey)
+	delegatorPrefixKey := GetDelegationsKey(delegatorAddr)
+	iterator := sdk.KVStorePrefixIterator(store, delegatorPrefixKey) //smallest to largest
+	defer iterator.Close()
+
+	i := 0
+	for ; iterator.Valid() && i < int(maxRetrieve); iterator.Next() {
+		addr := iterator.Key()
+		delegation := types.MustUnmarshalDelegation(k.cdc, addr, iterator.Value())
+
+		validator, found := k.GetValidator(ctx, delegation.ValidatorAddr)
+		if !found {
+			panic(types.ErrNoValidatorFound(types.DefaultCodespace))
+		}
+		validators[i] = validator
+		i++
+	}
+	return validators[:i] // trim
+}
+
+// return a validator that a delegator is bonded to
+func (k Keeper) GetDelegatorValidator(ctx sdk.Context, delegatorAddr sdk.AccAddress,
+	validatorAddr sdk.ValAddress) (validator types.Validator, err sdk.Error) {
+
+	delegation, found := k.GetDelegation(ctx, delegatorAddr, validatorAddr)
+	if !found {
+		return validator, types.ErrNoDelegation(types.DefaultCodespace)
+	}
+
+	validator, found = k.GetValidator(ctx, delegation.ValidatorAddr)
+	if !found {
+		panic(types.ErrNoValidatorFound(types.DefaultCodespace))
+	}
+	return
+}
+
+//_____________________________________________________________________________________
+
+// return all delegations for a delegator
+func (k Keeper) GetAllDelegatorDelegations(ctx sdk.Context, delegator sdk.AccAddress) (
+	delegations []types.Delegation) {
+
+	store := ctx.KVStore(k.storeKey)
+	delegatorPrefixKey := GetDelegationsKey(delegator)
+	iterator := sdk.KVStorePrefixIterator(store, delegatorPrefixKey) //smallest to largest
```

Comment:
I think these comments of ` //smallest to largest` are redundant and can be removed unless you actually specify what its ranging on.